pkg/gogen/templates.go

```diff
@@ -112,6 +114,47 @@ func (t *{{$Md.Name}}) SetDefaults() {
 	{{- end }}
 }
 
+// {{$Md.Name}}Publisher wraps rclgo.Publisher to provide type safe helper
+// functions
+type {{$Md.Name}}Publisher struct {
+	*rclgo.Publisher
+}
+
+// New{{$Md.Name}}Publisher creates and returns a new publisher for the
+// {{$Md.Name}}
+func New{{$Md.Name}}Publisher(node *rclgo.Node, topic_name string, options *rclgo.PublisherOptions) (*{{$Md.Name}}Publisher, error) {
+	pub, err := node.NewPublisher(topic_name, {{$Md.Name}}TypeSupport, options)
+	if err != nil {
+		return nil, err
+	}
+	return &{{$Md.Name}}Publisher{pub}, nil
+}
+
+func (p *{{$Md.Name}}Publisher) Publish(msg *{{$Md.Name}}) error {
+	return p.Publisher.Publish(msg)
+}
+
+// {{$Md.Name}}Subscription wraps rclgo.Subscription to provide type safe helper
+// functions
+type {{$Md.Name}}Subscription struct {
+	*rclgo.Subscription
+}
+
+// New{{$Md.Name}}Subscription creates and returns a new subscription for the
+// {{$Md.Name}}
+func New{{$Md.Name}}Subscription(node *rclgo.Node, topic_name string, subscriptionCallback rclgo.SubscriptionCallback) (*{{$Md.Name}}Subscription, error) {
+	sub, err := node.NewSubscription(topic_name, {{$Md.Name}}TypeSupport, subscriptionCallback)
+	if err != nil {
+		return nil, err
+	}
+	return &{{$Md.Name}}Subscription{sub}, nil
+}
```

Comment:
Would be nice to be able to just use API as this:
```
NewFooSubscription(node, "topic", func(msg Foo, info rclgo.RmwMessageInfo) {
    fmt.Printf("got message: %+v", msg)
})
```

```suggestion

type {{$Md.Name}}SubscriptionCallback func(msg {{$Md.Name}}, info rclgo.RmwMessageInfo)

// New{{$Md.Name}}Subscription creates and returns a new subscription for the
// {{$Md.Name}}
func New{{$Md.Name}}Subscription(node *rclgo.Node, topic_name string, subscriptionCallback {{$Md.Name}}SubscriptionCallback) (*{{$Md.Name}}Subscription, error) {

	callback := func(s *rclgo.Subscription) {
		var msg {{$Md.Name}}
		err, info := s.TakeMessage(msg)
		// TODO: handle error?
		subscriptionCallback(msg, info)
	}

	sub, err := node.NewSubscription(topic_name, {{$Md.Name}}TypeSupport, callback)
	if err != nil {
		return nil, err
	}
	return &{{$Md.Name}}Subscription{sub}, nil
}
```

Reply:
The error could be passed to the handler to let the user decide how to handle the error.

```suggestion
type {{$Md.Name}}SubscriptionCallback func(msg *{{$Md.Name}}, info *rclgo.RmwMessageInfo, err error)

// New{{$Md.Name}}Subscription creates and returns a new subscription for the
// {{$Md.Name}}
func New{{$Md.Name}}Subscription(node *rclgo.Node, topic_name string, subscriptionCallback {{$Md.Name}}SubscriptionCallback) (*{{$Md.Name}}Subscription, error) {
	callback := func(s *rclgo.Subscription) {
		var msg {{$Md.Name}}
		info, err := s.TakeMessage(&msg)
		subscriptionCallback(&msg, info, err)
	}
	sub, err := node.NewSubscription(topic_name, {{$Md.Name}}TypeSupport, callback)
	if err != nil {
		return nil, err
	}
	return &{{$Md.Name}}Subscription{sub}, nil
}
```

Reply:
`{type}SubscriptionCallback` function type added and implemented according to the suggestion by @lassilaiho in commit: 3c63ebe.